src/higherOrders/withApolloClient/initApollo.js

```diff
@@ -3,35 +3,46 @@
 import { ApolloClient, HttpLink, InMemoryCache } from 'apollo-boost'
 import fetch from 'isomorphic-unfetch'
 import getAccessToken from '@helpers/getAccessToken'
+import clearAccessToken from '@helpers/clearAccessToken'
+import redirectTo from '@helpers/redirectTo'
 
 let apolloClient = null
 
-const customFetch = (URI, options) => {
+const customFetch = (context: any) => async (URI: string, options: any) => {
   const token = getAccessToken()
   options.headers.Authorization = token ? `bearer ${token}` : ''
 
-  return fetch(URI, options)
+  const response = await fetch(URI, options)
+
+  // We intercept the response and log the user out if their access token is not
+  // valid.
+  if (response.status === 401) {
```

Comment:
I don’t think we throw a 401 for other reasons and we probably shouldn’t :)

Reply:
👍